pkg/controller/registry/reconciler/grpc.go

```diff
@@ -327,3 +324,34 @@ func (c *GrpcRegistryReconciler) CheckRegistryServer(catalogSource *v1alpha1.Cat
 	healthy = true
 	return
 }
+
+// promoteCatalog swaps the labels on the update pod so that the update pod is now reachable by the catalog service.
+// By updating the catalog on cluster it promotes the update pod to act as the new version of the catalog on-cluster.
+func (c *GrpcRegistryReconciler) promoteCatalog(updatePod *corev1.Pod, key string) error {
+	// Update the update pod to promote it to serving pod via the SSA client
+	err := c.SSAClient.Apply(context.TODO(), updatePod, func(p *v1.Pod) error {
+		p.Labels[CatalogSourceLabelKey] = key
+		p.Labels[CatalogSourceUpdateKey] = ""
+		return nil
+	})()
+
+	return err
+}
+
+// containerReady returns true if the given Pod has a ready status for its container.
+func containerReady(pod *corev1.Pod) bool {
+	if pod.Status.ContainerStatuses != nil {
```

Comment:
I'm a bit confused by this as if there is no container statuses (== nil), should we consider it as NotReady?

Reply:
I also wonder if we need to check `containerStatuses` for readiness reason. We technically care if the pod is ready. So maybe we should check for `Status.Conditions` for the` type: Ready` as `true`. We only care about container for its `imageID`. This way we won't need to change the code if the pod contains multiple containers in the future.

Reply:
We don't care about the pod as much as the container in the pod but I agree this reasoning is easier to follow. Will update.